streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -787,9 +796,12 @@ public DefaultStateUpdater(final String name,
         this.log = logContext.logger(DefaultStateUpdater.class);
     }
 
-    @Override
     public void start() {
         if (stateUpdaterThread == null) {
+            if (!restoredActiveTasks.isEmpty() || !exceptionsAndFailedTasks.isEmpty()) {
```

Comment:
I thought that since we allow to restart the state updater, we should verify that the state updater starts with clean queue to avoid invalid states coming from a past run of the state updater. At the moment, we never  restart the state updater but I thought having clear invariants might be good.

Reply:
I wonder why we'd even allow restarting the state updater. but we don't need to fix it now